Removing `glyph.changed()` from `scalePoints`

colinmford opened the pull request:
Hi @klaavo,

On the RoboFont Discord I was talking with @typemytype investigating a slow-down with ScalingEditTool reported by @jeremymickel. I took a look and it seems like `glyph.changed()` was being called every `mouseDragged` event (line 252), which is a pretty expensive operation to call many times a second. Removing it increased the tool's responsiveness. As far as I could see it was a redundant call, but if there was a specific reason for it please let me know and maybe we can restructure it to just get called less often.

While I was in there I also noticed the logic of Line 171 wasn't making sense. Based on your very helpful comments I added some parens to clean up the logic of that if statement.

Thanks!

Colin

colinmford committed:
Removing `glyph.changed()` from `scalePoints`

klaavo commented:
Hi Colin!
Thanks for your input! As I wrote on discord, we still need to update the selection highlight somehow, but there is probably a better way than calling glyph.changed(). I cannot reproduce any major slow down.
The parens for the conditional is good for readability, but now when I think of it, not self.isDragging() or event.keyCode() == 48: should be enough for that elif.
Now I'm also trying to remember why the modifiersChanged calls buildScaleDataList on line 165. It seems redundant.

colinmford committed:
Incorporating feedback

colinmford commented on ScalingEditTool.roboFontExt/lib/scalingEditToolExt.py:
```diff
@@ -248,8 +250,10 @@ def scalePoints(self, arrowKeyDown=0):
+            
+            # Fredrik's alternative to glyph.changed()
+            glyph.asDefcon().selection.resetSelectionPath()
+            glyph.asDefcon().selection.dirty = True
```
Incorporated @typemytype's suggestion from Discord, I think it works!

colinmford commented:
Hey @klaavo,

I incorporated @typemytype's suggestion from Discord, and simplified line 172 per your comment! I left line 165 as-is for now. It didn't seem like it was slowing anything down.

colinmford commented:
Thanks so much, @klaavo!
